frontend/src/components/atoms/ProjectionRowEntry.tsx

```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components';
+
+type ITransaction = import('../../declarations/transaction').ITransaction;
+
+interface IProjectionRowEntryProps {
+  readonly type: ITransaction['type'] | undefined;
+  readonly gapAbove: boolean;
+}
+
+export const ProjectionRowEntry = styled.div<IProjectionRowEntryProps>`
```

Comment:
Looks way better!

One final note, to avoid having a nullable type (ref. `ITransaction['type'] | undefined`) prop on all ProjectionRowEntries here, you could extract the header to it's own component, thus removing the need for a css `className` (and using css class styling, as we're trying to avoid that to the best of our abilities).

The code for this would look something like this;

```tsx
const ProjectionRowHeader = styled.div`
  /* The original styling from the "projection-header" class */
`;
```

I don't know if there's much overlap between the two classes, but if there is, there is an option for that in styled-components;

```tsx
const ProjectionRow = styled.div`
 /* styling that is common for both header and entry ... */
`;

const ProjectionRowEntry = styled<IProjectionRowEntryProps>(ProjectionRow)`
  /* styling for entry here */
`;

const ProjectionRowHeader = styled(ProjectionRow)`
  /* styling for header here */
`;
```

Reply:
This is called extending styled components in case you want to google it.

https://www.styled-components.com/docs/api#using-custom-props